source/numerics/error_estimator.cc

```diff
@@ -464,6 +464,98 @@ namespace internal
       return face_integral;
     }
 
+    /**
+     * A factor to scale the integral for the face at the boundary.
+     * Used for Neumann BC (?).
+     */
+    template <class DH>
+    double regular_face_factor_boundary(const typename DH::active_cell_iterator &cell,
+                                        const unsigned int                       face_no,
+                                        const dealii::hp::FEFaceValues<DH::dimension, DH::space_dimension> &fe_face_values_cell,
+                                        typename KellyErrorEstimator<DH::dimension,DH::space_dimension>::Strategy strategy)
+    {
+      switch (strategy)
+        {
+        case KellyErrorEstimator<DH::dimension,DH::space_dimension>::Ainsworth :
+          return 1.0;
+          break;
+        case KellyErrorEstimator<DH::dimension,DH::space_dimension>::face_diameter_over_twice_max_degree :
+          const double cell_degree = fe_face_values_cell.get_fe_collection()[cell->active_fe_index()].degree;
+          return cell->face(face_no)->diameter() / cell_degree;
```

Comment:
what degree do you take if this and the neighboring face have different polynomial degrees?